src/components/Footer.tsx

```diff
@@ -0,0 +1,63 @@
+import { FC, useState, useEffect } from 'react';
+import { Todo } from '../types/Todo';
+import { TodosFilter } from './TodosFilter';
+
+type Props = {
+  todos: Todo[],
+  onTodoDelete: (todoIds: number[]) => void;
+};
+
+export const Footer: FC<Props> = ({ todos, onTodoDelete }) => {
+  const [activeTodosCount, setActiveTodosCount] = useState(0);
+  const [completedTodosCount, setCompletedTodosCount] = useState(0);
+  const [completedTodos, setCompletedTodos] = useState<Todo[]>([]);
+
+  const handleCompletedTodos = () => {
+    const doneTodos = todos.filter(todo => todo.completed);
+
+    setCompletedTodosCount(doneTodos.length);
+    setCompletedTodos(doneTodos);
+  };
+
+  const handleClearCompleted = () => {
+    const todoIdsToDelete = completedTodos.map(todo => todo.id);
+
+    onTodoDelete(todoIdsToDelete);
+  };
+
+  const handleActiveTodosCount = () => {
+    const activeCount = todos.filter(todo => !todo.completed).length;
+
+    setActiveTodosCount(activeCount);
+  };
+
+  useEffect(() => {
+    handleActiveTodosCount();
+    handleCompletedTodos();
+  }, [todos]);
```

Comment:
Looks strange a bit. Do you really need states for the values above and set them from the useEffect? Wouldn't it be better to use just useMemo() for these values? 🤔